service/crt/decoder.go

```diff
@@ -15,6 +16,14 @@ type certificateDecoder struct {
 	close   func() error
```

Comment:
Nitpicking: we can have a stream there and close it explicitly. I don't see a benefit of hiding it behind a function. If for whatever reason you want to hide Read method, you can use io.Closer type here.